templates/terraform/resource.erb

```diff
@@ -170,25 +170,25 @@ func resource<%= resource_name -%>Update(d *schema.ResourceData, meta interface{
 
 	d.Partial(true)
 
-	<% settable_properties.reject { |p| p.update_url.nil? }.each do |prop| -%>
-	if d.HasChange("<%= Google::StringUtils.underscore(prop.name) -%>") {
-		<% settable_properties.each do |prop| -%>
+	<% properties_by_custom_update(settable_properties).each do |props| -%>
+	if <%= props.map { |prop| "d.HasChange(\"#{Google::StringUtils.underscore(prop.name)}\")" }.join ' || ' -%> {
+		<% props.each do |prop| -%>
 			<%= prop.api_name -%>Prop, err := expand<%= resource_name -%><%= titlelize_property(prop) -%>(d.Get("<%= Google::StringUtils.underscore(prop.name) -%>"), d, config)
 			if err != nil {
 				return err
 			}
 		<% end -%>
 
 		obj := map[string]interface{}{
-		<% settable_properties.each do |prop| -%>
+		<% props.each do |prop| -%>
 			"<%= prop.api_name -%>": <%= prop.api_name -%>Prop,
 		<% end -%>
 		}
-		url, err = replaceVars(d, config, "<%= update_url(object, prop.update_url) -%>")
+		url, err = replaceVars(d, config, "<%= update_url(object, props[0].update_url) -%>")
```

Comment:
This would be an antipattern in python- can you get a ruby person to approve it?

Reply:
@nelsonjr I could do `props.last`. 

Context: The way `properties_by_custom_update`, we are guaranteed that there is one or more properties in props.

Reply:
I think neither. Because `properties_by_custom_update` uses `group_by` what you want is likely this:

```
<% properties_by_custom_update(settable_properties).each do |_, props| -%>
```

Why not instead of trying to use the values for this you do an actual map first and return the following:

```
properties_by_custom_update(settable_properties).each do |key, props|
  # keys is actually a hash: verb=>, update_url=>
  # ... 
  # then your code would be:

  update_url(object, key[:update_url])

```

Code would look cleaner IMHO.

It also seems you're ignoring the verb to pass to the update method. Different objects have different methods.